packages/block-editor/src/components/aspect-ratio-dropdown/index.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * WordPress dependencies
+ */
+import { check, aspectRatio as aspectRatioIcon } from '@wordpress/icons';
+import { DropdownMenu, MenuGroup, MenuItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useSettings } from '../use-settings';
+import { POPOVER_PROPS } from '../image-editor/constants';
+import { useImageEditingContext } from '../image-editor/context';
+
+function AspectRatioGroup( {
+	aspectRatios,
+	isDisabled,
+	label,
+	onClick,
+	value,
+} ) {
+	return (
+		<MenuGroup label={ label }>
+			{ aspectRatios.map( ( { name, slug, ratio } ) => (
+				<MenuItem
+					key={ slug }
+					disabled={ isDisabled }
+					onClick={ () => {
+						onClick( ratio );
+					} }
+					role="menuitemradio"
+					isSelected={ ratio === value }
+					icon={ ratio === value ? check : undefined }
+				>
+					{ name }
+				</MenuItem>
+			) ) }
+		</MenuGroup>
+	);
+}
+
+function ratioToNumber( str ) {
+	// TODO: support two-value aspect ratio?
+	// https://css-tricks.com/almanac/properties/a/aspect-ratio/#aa-it-can-take-two-values
+	const [ a, b, ...rest ] = str.split( '/' ).map( Number );
+	if (
+		a <= 0 ||
+		b <= 0 ||
+		Number.isNaN( a ) ||
+		Number.isNaN( b ) ||
+		rest.length
+	) {
+		return NaN;
+	}
+	return b ? a / b : a;
+}
+
+function presetRatioAsNumber( { ratio, ...rest } ) {
+	return {
+		ratio: ratioToNumber( ratio ),
+		...rest,
+	};
+}
```

Comment:
Having unit tests for these utils would be helpful.

Reply:
Added a test in https://github.com/WordPress/gutenberg/pull/47271/commits/18c204517ea7f6f82b416f2e56f00355216fcbb8